src/core/services/CreditLineService.ts

```diff
@@ -1,46 +1,220 @@
+    return await this.executeContractMethod('increaseCredit', [id, amount], dryRun);
+  }
+
+  public async depositAndRepay(
+    amount: BigNumberish,
+    dryRun: boolean
+  ): Promise<TransactionResponse | PopulatedTransaction> {
+    return await this.executeContractMethod('depositAndRepay', [amount], dryRun);
+  }
+
+  public async depositAndClose(dryRun: boolean): Promise<TransactionResponse | PopulatedTransaction> {
+    return await this.executeContractMethod('depositAndClose', [], dryRun);
+  }
+
+  private async executeContractMethod(methodName: string, params: any[], dryRun: boolean) {
+    let props: ExecuteTransactionProps | undefined = undefined;
+    try {
+      props = {
+        network: 'mainnet',
+        args: params,
+        methodName: methodName,
+        abi: this.abi,
+        contractAddress: this.contractAddress,
+      };
+      if (dryRun) {
+        return await this.transactionService.populateTransaction(props);
+      }
+
+      let tx;
+      tx = await this.transactionService.execute(props);
+      await tx.wait();
+      return tx;
+    } catch (e) {
+      console.log(
+        `An error occured while ${methodName} with params [${params}] on CreditLine [${
+          props?.contractAddress
+        }], error = [${JSON.stringify(e)}] `
+      );
+      return Promise.reject(e);
+    }
+  }
+
+  /* ============================= Helpers =============================*/
+
+  public async getLenderByCreditID(id: BytesLike): Promise<Address> {
+    return (await this.contract.credits(id)).lender;
+  }
+
+  public async getFirstID(): Promise<BytesLike> {
+    return await this.contract.ids(0);
+  }
+
+  public async getCredit(id: BytesLike): Promise<Credit> {
+    return await this.contract.credits(id);
+  }
+
+  public async borrower(): Promise<Address> {
+    return await this.contract.borrower();
+  }
+
+  public async isActive(): Promise<boolean> {
+    return (await this.contract.status()) === STATUS.ACTIVE;
+  }
+
+  public async isBorrowing(): Promise<boolean> {
+    const id = await this.contract.ids(0);
+    return (await this.contract.count()) !== 0 && (await this.contract.credits(id)).principal !== 0;
+  }
+
+  public async isBorrower(): Promise<boolean> {
+    return (await this.getSignerAddress()) === (await this.contract.borrower());
+  }
+
+  public async isLender(id: BytesLike): Promise<boolean> {
+    return (await this.getSignerAddress()) === (await this.getLenderByCreditID(id));
+  }
+
+  public async isMutualConsent(trxData: string | undefined, signerOne: Address, signerTwo: Address): Promise<boolean> {
+    const signer = await this.getSignerAddress();
+    const isSignerValid = signer === signerOne || signer === signerTwo;
+    const nonCaller = signer === signerOne ? signerTwo : signerOne;
+    const expectedHash = keccak256(ethers.utils.solidityPack(['string', 'address'], [trxData, nonCaller]));
+    return isSignerValid && this.contract.mutualConsents(expectedHash);
```

Comment:
doesnt `&& this.contract.mutualConsents(expectedHash)`  mean no tx will ever be initiated from our site because all the functions currently require this to return true before sendinga  tx but a tx must be sent for it to be true?